diff --git a/node_modules/hmc-win32/hmc.d.ts b/node_modules/hmc-win32/hmc.d.ts
index 9d1502b..5ca0db2 100644
--- a/node_modules/hmc-win32/hmc.d.ts
+++ b/node_modules/hmc-win32/hmc.d.ts
@@ -3212,6 +3212,7 @@ export declare const hmc: {
         sequence: typeof getClipboardSequenceNumber;
         watch: typeof watchClipboard;
     };
+    // @ts-ignore
     HMC: typeof HMC;
     HWND: typeof HWND;
     MessageError: typeof MessageError;
